pkg/chunk/cache/redis_client.go

```diff
@@ -110,17 +110,34 @@ func (c *RedisClient) MGet(ctx context.Context, keys []string) ([][]byte, error)
 		defer cancel()
 	}
 
-	cmd := c.rdb.MGet(ctx, keys...)
-	if err := cmd.Err(); err != nil {
-		return nil, err
-	}
-
 	ret := make([][]byte, len(keys))
-	for i, val := range cmd.Val() {
-		if val != nil {
-			ret[i] = StringToBytes(val.(string))
+
+	// redis.UniversalClient can take redis.Client and redis.ClusterClient.
+	// if redis.Client is set, then Single node or sentinel configuration. mget is always supported.
+	// if redis.ClusterClient is set, then Redis Cluster configuration. mget may not be supported.
+	_, isCluster := c.rdb.(*redis.ClusterClient)
+
+	if isCluster {
+		for i, key := range keys {
+			cmd := c.rdb.Get(ctx, key)
+			if err := cmd.Err(); err != nil {
+				return nil, err
+			}
+			ret[i] = StringToBytes(cmd.Val())
```

Comment:
If a key doesn't exist, then `ret[i]` is expected to be `nil`. I don't think it's the case here.

Reply:
@pracucci thanks for reviewing! `c.rdb.MGet(...).Val()` returns `[]interface{}`, but `c.rdb.Get(...).Val()` returns string. If the value is not found, it will return an empty string.

Reply:
> If the value is not found, it will return an empty string.

I think this break the behaviour (see my previous comment above). Having a unit test to prove / disprove it would be great.

Reply:
understood. Thank you!

Reply:
Processed the empty values and added the tests. The tests confirm that the behavior is the same for Single and Cluster configurations of Redis.